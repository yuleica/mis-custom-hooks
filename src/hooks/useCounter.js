import { useState } from 'react';

export const useCounter = (initialState = 1) => {

    const [counter, setcounter] = useState(initialState);

   const increment = (factor = 1) => {
       setcounter(counter + factor);
    }


    const decrement = (factor = 1) => {
        setcounter(counter - factor);
    }

    const reset = () => setcounter(initialState);

    return{
        counter,
        increment,
        decrement,
        reset
    };

}
