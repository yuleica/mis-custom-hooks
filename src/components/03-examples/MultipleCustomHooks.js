import React from 'react';
import { useFetch } from '../../hooks/useFetch';
import {useCounter} from '../../hooks/useCounter';

import '../02-useEffect/effects.css';


export const MultipleCustomHooks = () => {
   
    const {counter, increment } = useCounter();

    const {loading, data} = useFetch(`https://breakingbadapi.com/api/quotes/${counter}`);

//equivale a un if. !!data=false
//si existe la data produce la carga la posicion 0 de la data
//si no existe, no eval√∫a y ambas variables son undefined 
    const {author, quote} = !!data && data[0];


   return(
       <div>
           <h1>Breaking Bad Quotes</h1>
           <hr />

           {
               loading
               ?
               (
                    <div className="alert alert-info text-center">
                    loading...
                    </div>
               )
               :
               (
                    <blockquote className="blockquote">
                        <p className="mb-0">{quote}</p>
                        <footer className="blockquote-footer text-right">{author}</footer>
                    </blockquote>
               )
           }

            <button className="btn btn-primary"
                    onClick={() => increment(1)}>
                    Siguiente quote
            </button>


       </div>
   )

}
