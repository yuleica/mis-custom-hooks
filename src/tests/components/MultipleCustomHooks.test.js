import { shallow } from 'enzyme';
import React from 'react';
import { MultipleCustomHooks } from '../../components/03-examples/MultipleCustomHooks';

//import {useCounter} from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch';


jest.mock('../../hooks/useFetch');
//jest.mock('../../hooks/useCounter');


test('should show correctly', () => {

//    useCounter.mockReturnValue({
//        counter: 1,
//        increment: () => {
//            return counter+1;
//        }
//    });
    
    useFetch.mockReturnValue({
        data: null,
        loading: true,
        error: null
    });

//debo pasar 1 como argumento en increment.. favor revisar las otras pruebas que hagan algo similar
 //   useCounter.mockReturnValue({
 //       counter: 1,
 //       increment()
//
//    });
        
    
    const wrapper = shallow(<MultipleCustomHooks />);

    expect (wrapper).toMatchSnapshot();
})



test('should show info', () => {

    useFetch.mockReturnValue({
        data: [{
            author: 'x en la vida',
            quote: 'hola mundo'

        }], 
        loading: false,
        error: null
    });


    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper.find('.alert').exists()).toBe(false);
    expect(wrapper.find('.mb-0').text().trim()).toBe('hola mundo');
    expect(wrapper.find('footer').text().trim()).toBe('x en la vida');
   
    console.log(wrapper.html())


})