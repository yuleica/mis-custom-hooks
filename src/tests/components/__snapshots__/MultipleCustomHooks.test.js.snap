// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show correctly 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;
